### PR TITLE
Lithium: cache chunk gen sea level

### PR DESCRIPTION
**Pull request by Titaniumtown**
This ports the optimization from: https://github.com/jellysquid3/lithium-fabric/pull/179 (just recently merged)
Making it so the sea level chunk gen setting is cached instead of calling for it `every single time a block is placed by the generator, for a total of 65536 registry lookups per chunk`. While this isn't a huge performance boost, it's still there. SuperCoder7979 said that `13% of CPU time spent in populateNoise without the patch vs 10% with the patch`

### CODE REVIEW

File: patches/server/0066-lithium-cache-chunk-gen-sea-level.patch [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
++     * Use cached sea level instead of retrieving from the registry every time.
++     * This method is called for every block in the chunk so this will save a lot of registry lookups.
++     *
++     * @author SuperCoder79
++     */
+     @Override
+     public int getSeaLevel() {
+-        return ((GeneratorSettingBase) this.h.get()).g();
++        if (this.cachedSeaLevel == Integer.MIN_VALUE) {
++            this.cachedSeaLevel = ((GeneratorSettingBase) this.h.get()).g();
```
Comment (MrIvanPlays) [inline patches/server/0066-lithium-cache-chunk-gen-sea-level.patch]: obfhelpers

**Review by MrIvanPlays — DISMISSED** [review]

File: patches/server/0066-lithium-cache-chunk-gen-sea-level.patch (hunk 1 above)
Comment (Titaniumtown) [reply patches/server/0066-lithium-cache-chunk-gen-sea-level.patch]: Fixed

Comment (Titaniumtown) [reply patches/server/0066-lithium-cache-chunk-gen-sea-level.patch]: I did the obfhelper wrong lol. I'll fix it shortly.

Comment (Titaniumtown) [reply patches/server/0066-lithium-cache-chunk-gen-sea-level.patch]: Turns out adding an obfhelper in `GeneratorSettingBase.java` screws up everything, so I'm just not going to add one lol.

**Comment by Titaniumtown** [comment]
Note: The PR to lithium was recently merged. Any more thoughts?

File: patches/server/0067-lithium-cache-chunk-gen-sea-level.patch [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
++     * Use cached sea level instead of retrieving from the registry every time.
++     * This method is called for every block in the chunk so this will save a lot of registry lookups.
++     *
++     * @author SuperCoder79
++     */
+     @Override
+-    public int getSeaLevel() {
+-        return ((GeneratorSettingBase) this.h.get()).g();
++    public int getSeaLevel() { // Cached method of getting sealevel
++        if (this.cachedSeaLevel == Integer.MIN_VALUE) this.cachedSeaLevel = ((GeneratorSettingBase) this.h.get()).g();
```
Comment (ishland) [inline patches/server/0067-lithium-cache-chunk-gen-sea-level.patch]: obfhelpers

**Review by ishland — DISMISSED** [review]

File: patches/server/0067-lithium-cache-chunk-gen-sea-level.patch (hunk 2 above)
Comment (Titaniumtown) [reply patches/server/0067-lithium-cache-chunk-gen-sea-level.patch]: > Turns out adding an obfhelper in `GeneratorSettingBase.java` screws up everything, so I'm just not going to add one lol.

**Comment by Titaniumtown** [comment]
Seems they changed it before pushing to lithium, so I have to update the patch: https://github.com/jellysquid3/lithium-fabric/commit/a55cfd1c914789a27bb6fe20e6a56af44d305f72

**Review by ishland — APPROVED** [review]

**Pull request merged**